IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs

```diff
@@ -0,0 +1,281 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.IE;
+using OpenQA.Selenium.Remote;
+using OpenQA.Selenium.Support.UI;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace IntelliTect.TestTools.Selenate
+{
+    public enum BrowserType
+    {
+        Chrome,
+        InternetExplorer
+    }
+    public class Browser
+    {
+        public Browser(BrowserType browser)
+        {
+            InitDriver(browser);
+        }
+
+        public IWebDriver Driver { get; set; }
+
+        // Mike C: Find a good way to abstract this out. Different projects will have different requirements here.
+        // Good candidate for an extension? Or maybe an abstract class?
+        public void InitDriver(BrowserType browser)
+        {
+            Driver?.Quit();
+
+            switch (browser)
+            {
+                case (BrowserType.Chrome):
+                    ChromeOptions chromeOptions = new ChromeOptions();
+                    chromeOptions.AddArgument("--disable-extension");
+                    chromeOptions.AddArgument("no-sandbox");
+                    chromeOptions.AddArgument("disable-infobars");
+                    chromeOptions.AddUserProfilePreference("credentials_enable_service", false);
+                    chromeOptions.AddUserProfilePreference("profile.password_manager_enabled", false);
+                    Driver = new ChromeDriver(Directory.GetCurrentDirectory(), chromeOptions, TimeSpan.FromMinutes(1));
+
+                    break;
+                case (BrowserType.InternetExplorer):
+                    InternetExplorerOptions ieCaps = new InternetExplorerOptions
+                    {
+                        EnablePersistentHover = true,
+                        EnsureCleanSession = true,
+                        EnableNativeEvents = true,
+                        IgnoreZoomLevel = true,
+                        IntroduceInstabilityByIgnoringProtectedModeSettings = true,
+                        RequireWindowFocus = false
+                    };
+                    Driver = new InternetExplorerDriver(ieCaps);
+                    break;
+                default:
+                    throw new Exception($"Unknown browser: {browser}");
```

Comment:
Given that this exception is being thrown because of the passed in argument value. It would be better to throw an ArgumentException. Or possibly an ArgumentOutOfRangeException

Reply:
Fixed with next commit